src/eos-updater-util.c

```diff
@@ -106,6 +107,48 @@ eos_updater_local_repo (void)
   return ret;
 }
 
+static gchar *
+cleanstr (gchar *s)
+{
+  gchar *i, *j, *k;
+
+  if (!s)
+    return s;
+
+  for (i = s; *i; i++)
+    /* only allow printable */
+    if (*i < 32 || *i > 126)
```

Comment:
I didn't look at what range does `g_ascii_isprint()` considers as printable chars, but many definitions of it I found in the web says DEL (127) is also part of the printable range, and we do not want that as part of our query/key. I like having it explicit in this case.